app/controllers/UserController.py

```diff
@@ -0,0 +1,196 @@
+        email = request.form.get('email')
+        password = request.form.get('password')
+        school = request.form.get('school')
+        role = request.form.get('role')
+        grade = request.form.get('grade')
+        updated_at = datetime.utcnow
+
+        data = [
+            {
+                'name': name,
+                'email': email,
+                'password': password,
+                'school': school,
+                'role': role,
+                'grade': grade,
+                'updated_at': updated_at
+            }
+        ]
+        
+        user = User.query.filter_by(id=id).first()
+
+        if user:
+            if name is not None:
+                user.name = name
+            if email is not None:
+                user.email = email
+            if password is not None:
+                user.password = password
+                user.setPassword(user.password)
+            if school is not None:
+                user.school = school
+            if role is not None:
+                user.role = role
+            if grade is not None:
+                user.grade = grade
+
+            db.session.commit()
+
+            return 'Data user has been updated'
+
+        else:
+            return 'Update data failed'
+    
+    except Exception as e:
+        print(e)
+
+def deleteUser(id):
+    try:
+        user = User.query.filter_by(id=id).first()
+        if not user:
+            return response.badRequest([], 'User not found')
+        db.session.delete(user)
+        db.session.commit()
+
+        return response.success('', 'Data user has been deleted')
+    
+    except Exception as e:
+        print(e)
+
+def userData(data):
+    data = {
+        'id': data.id,
+        'name': data.name,
+        'email': data.email,
+        'school': data.school,
+        'grade': data.grade,
+        'role': data.role,
+        'dob': data.dob,
+    }
+
+    return data
+
+def login():
+    try:
+        email = request.form.get('email')
+        password = request.form.get('password')
+
+        user = User.query.filter_by(email=email).first()
+
+        if not user:
+            return response.badRequest([], 'email or password are incorrect')
+        
+        if not user.checkPassword(password):
+            return response.badRequest([], 'email or password are incorrect')
+        
+        data = userData(user)
+
+        expires = datetime.timedelta(days=7)
+        expires_refresh = datetime.timedelta(days=10)
+
+        access_token = create_access_token(data, fresh=True, expires_delta= expires)
+        refresh_token = create_refresh_token(data, expires_delta=expires_refresh)
+
+        return response.success({
+            'data': data,
+            'access_token': access_token,
+            'refresh_token': refresh_token
+        }, 'Login Success')
+
+    except Exception as e:
+        print(e)
+
+def logout():
+    try:
+        # Ambil identitas (biasanya berisi informasi pengguna)
```

Comment:
we can remove the comments since the code is readable enough

Reply:
done